src/plot.jl

```diff
@@ -0,0 +1,448 @@
+    categories = loginfo.categories
+    hasdefault = (loginfo.timespercat[end] !=0 ? true : false) 
+
+    # Axis attributes
+    lengthx = length(categories)
+    ticks = categories
+    hasdefault && (lengthx += 1)
+    hasdefault && (ticks = [ticks..., "default"])
+    ax.xticks = (1:lengthx, ticks)
+    ax.ylabel = "Time (s)"
+
+
+    # Colors
+    grad = cgrad(:tab10)[1:length(categories)]
+    hasdefault && (grad = [grad..., :black])
+
+    # Barplot
+    # -------
+    y = loginfo.timespercat[1:end]
+    !hasdefault && (y = y[1:end-1])
+    barplot!(
+        ax,
+        1:lengthx,
+        y,
+        color = grad
+    )
+
+end
+
+
+"""
+    dagplot(ax, logger)  
+Plot the dag to the Makie axis ax
+"""
+function dagplot(ax, logger)
+    # Create GraphViz graph DOT format file
+    g = GraphViz.Graph(loggertodot(logger))
+
+    # Node positionning
+    GraphViz.layout!(g)
+
+    # Render this to a Cairo context to convert in png
+    cs = GraphViz.cairo_render(g)
+    Cairo.write_to_png(cs, "./dag_tmp.png")
+
+    # Load the png we juste created
+    img = load("./dag_tmp.png")
+
+    # Render the png on the Makie axis
+    GLMakie.image!(ax, rotr90(img))
+
+    # Axis attributes
+    ax.aspect = DataAspect()
+    hidedecorations!(ax)
+    hidespines!(ax)
+
+    # Remove the temporary png
+    rm("./dag_tmp.png")
+end
+function dagplot(logger=getlogger())
+    fig = Figure()
+    dagplot(Axis(fig[1,1], title = "Graph"), logger)
+    fig
+end
+
+
+function react(ax, logger::Logger, gantt::Gantt)
+    to = Observable("")
+
+    on(events(ax.parent).mouseposition) do mp
+        pos = mouseposition(ax.scene)
+
+        k = 1
+        match = false
+        for tasklog ∈ Iterators.flatten(logger.tasklogs)
+            condx = gantt.starts[k] <= pos[1] <= gantt.stops[k]
+            condy = gantt.threads[k] - 0.3 <= pos[2] <= gantt.threads[k] + 0.3
+            if condx && condy
+                match = true
+                tasklog.label != "" && (to[] = "$(tasklog.label)")
+            end
+
+            k += 1
+        end
+
+        !match && (to[] = "Task Label")
+
+    end
+    on(to) do t
+        ax.title = "Parallel Trace\n$t"
+    end
+end
+
+
+"""
+    plot(logger; categories)  
+Plot DataFlowTasks `logger` labeled informations with categories.  
+Note : if there's too many tasks, the DAG won't be plotted,
+although you can manually call `dagplot(logger)`.
+
+## Example  
+Use the above exemple with
+```jldoctest
+using GLMakie, GraphViz
+using DataFlowTasks
+using DataFlowTasks: RW
+import DataFlowTasks as DFT
+
+computing(A) = exp.(sum(A).^2).^2
+function work(A, B)
+    @dspawn computing(@RW(A)) label="A"
+    @dspawn computing(@RW(B)) label="B"
+    @dspawn computing(@RW(A)) label="A"
+    @dspawn computing(@RW(B)) label="B"
+end
+
+# Context
+A = ones(1000, 1000)
+B = ones(1000, 1000)
+
+# Compilation
+work(copy(A), copy(B))
+
+# Reset logger and taskcounter
+DFT.resetlogger!()
+DFT.TASKCOUNTER[] = 0
```

Comment:
Why two lines? Couldn't both these operations be realized by `resetlogger!`?

Reply:
> The two commits above:
> 
> 1. make sure the logging tests are actually run during CI tests (the file was originally not listed in `runtests.jl`)
> 2. avoid warnings about optional dependencies (such as `GLMakie`) being used in the source file but not listed in `Project.toml`. The way to do this is documented in `Requires.jl`'s [README](https://github.com/JuliaPackaging/Requires.jl):
>    > In the `@require` block, or any included files, you can use or import the package, but note that you must use the syntax `using .Gadfly` or `import .Gadfly`, rather than the usual syntax. Otherwise you will get a warning about `Gadfly` not being in dependencies.

Thanks for the tips, i planned on asking about that (didn't pay attention to this part of the Requires.jl doc it seems)